### DISCUSSION TITLE
Form submissions not working on subsequent submissions

### DISCUSSION THREAD

**Discussion by madison-swell** [swup/swup#866] [Help] [answered]
### Describe your problem 🧐

I have a default Wordpress search form in my header. 

The swup transition works perfectly on the first attempt. However any subsequent submissions do not and are redirected like a normal GET form.

I've disabled all my Wodpress plugins, other javascript, swup plugins and even tested with the default Wordpress TwentyTwentyFour theme.

This plugin works perfectly in CodeSandox so it must be something Wordpress is doing that's incompatible with the plugin.

I've narrowed it down to the 'form:submit' hook not being called on the subsequent attempt. 

I've noticed `beforeFormSubmit` is called successfully on subsequent submissions but the `swup.hooks.callSync('form:submit',..` (line 121 of index.ts) call is ineffective this time around. 

### What is your question? 🙋‍♀️

Any idea why this might be occurring and a possible fix?

### URL to minimal reproduction 🔗

N/A with codesandbox

### Checked all these? 📚

- [X] I have read and searched the [official docs](https://swup.js.org)
- [X] I have checked [discussions](https://github.com/orgs/swup/discussions) and [existing issues](https://github.com/issues?q=is:issue+org:swup) for answers to my question
- [X] I agree to follow this project's [Code of Conduct](https://github.com/swup/.github/blob/master/CODE_OF_CONDUCT.md)
- [X] I have provided all necessary information to the best of my knowledge

**Comment by daun** [comment]
I'd suspect the form element is being rendered differently when it was just submitted. Hard to say without an actual link – if you have a demo site, that'd be helpful. If, as you're saying, the method itself is called, but it doesn't end up at the bottom to call the actual hook, it might be any of these "abort" conditions that apply:

```ts
const opensInNewTabFromKeyPress = this.isSpecialKeyPressed();
const opensInNewTabFromTargetAttr = this.getFormAttr('target', form, submitter) === '_blank';
const opensInNewTab = opensInNewTabFromKeyPress || opensInNewTabFromTargetAttr;
```

**Comment by madison-swell** [comment]
Just working on getting a demo environment to show the behaviour.

In the meantime I've narrowed things down.

If I change line 129 of index.ts from:

```
swup.hooks.callSync('form:submit', { el: form, event }, () => {
    this.submitForm(event);
});
```

To simply:

```
this.submitForm(event);
```

Everything works perfect. 

For some reason the callSync method doesn't seem to do anything  after the first page load.

**Comment by madison-swell** [comment]
Demo wordpress site here:

https://swup.swelldev.com.au/?s=

As you can see – the first form submission works perfect. Subsequent submissions cause a page redirect.

This is a demo with the default wordpress theme. I can confirm I get exactly the same behaviour with my custom theme that uses different markup in the search form.

**Comment by daun** [comment]
Been doing some investigating — this seems to be an issue with the Forms Plugin itself.

I've created a [PR with a fix](https://github.com/swup/forms-plugin/pull/64) along with [a bug report](https://github.com/swup/forms-plugin/issues/63).

**Comment by daun** [comment, marked as answer]
This was indeed an issue with the plugin — we've improved how swup handles hooks in a previous release and haven't made some of the required adjustments to make the plugin compatible. This should be fixed in the latest plugin release [3.4.2](https://github.com/swup/forms-plugin/releases/tag/3.4.2).

**Reply by madison-swell** [reply]
I've upgraded to 3.4.2 and the problem is solved. Thanks so much!
